SConstruct

```diff
@@ -434,6 +415,22 @@ env.modules_detected = modules_detected
 opts.Update(env, {**ARGUMENTS, **env.Dictionary()})
 Help(opts.GenerateHelpText(env))
 
+
+# FIXME: Tool assignment happening at this stage is a direct consequence of getting the platform logic AFTER the SCons
+# environment was already been constructed. Fixing this would require a broader refactor where all options are setup
+# ahead of time with native validator/converter functions.
+tmppath = "./platform/" + env["platform"]
+sys.path.insert(0, tmppath)
+import detect
```

Comment:
This had to move up, otherwise `detect` here was actually the one of the last platform we checked when parsing all of them... i.e. windows, so Linux and macOS were getting `mingw` registered and ending with `.exe` binary extensions :p